### ISSUE TITLE
feature request: --ambiguous-bam

### ISSUE THREAD

**Issue by avilella** [FelixKrueger/Bismark#2]
An option to print out the ambiguously mapping reads as a separate bam file file, with MAPQ=0.

**Comment by FelixKrueger** [comment]
Added new option `--ambig_bam` in commit d6b7a82fefa05ddc3d04e15227731f2797b48cd7. The option works in Bowtie2-mode only, and works for single-end and paired-end alignments. Support for `--multicore` mode is yet to be added, but I thought you might want to have a play with it as it is and let me know if it works or is what you needed?

**Comment by avilella** [comment]
Brilliant! I am all over it, Felix! I will report back...

**Comment by avilella** [comment]
The ambig.bam file looks good, only the read ids have a `/1` and `/2` suffix which is not seen in the normal bam file. I presume these shouldn't be there:

```
D00476:188:C7UTCANXX:1:1210:10338:99473_1:N:0:CGATGTAT:ACAACT/1   99   chr1                        20770      1   126M           =  20780      136   TAATTAGTAG
D00476:188:C7UTCANXX:1:1210:10338:99473_2:N:0:CGATGTAT:ACAACT/2   147  chr1                        20780      1   126M           =  20770      -136  GGATTGTGTA
D00476:188:C7UTCANXX:1:1109:19480:3735_2:N:0:CGATGTAT:CAAATT/2    163  chr1                        47963      1   126M           =  48163      326   CAAATTATTT
D00476:188:C7UTCANXX:1:1109:19480:3735_1:N:0:CGATGTAT:CAAATT/1    83   chr1                        48163      1   126M           =  47963      -326  ATATTCACAT
```

**Comment by FelixKrueger** [comment]
`/1` and `/2` should no longer show in paired end ambiguous BAM files (fixed here e763a6883cab98697b8e3f32084a312e3b187894). If this works as expected I'll have a go at the multi-threading.

**Comment by FelixKrueger** [comment]
`--ambig_bam` is now also working in `--multicore` mode  (addressed in de4be9a802ba2c281f01ad40cc6c0cf37075bd55).
